contracts/DaiHrd.sol

```diff
@@ -464,4 +504,99 @@ contract DaiHrd is ERC777 {
 		vat.move(address(this), msg.sender, attorontodai);
 		return attorontodai;
 	}
+
+	// Dai specific functions. These functions all behave similar to standard functions with input or output denominated in Dai instead of DaiHrd
+	function balanceOfDai(address tokenHolder) external view returns(uint256 attodai) {
+		uint256 rontodaiPerPot = calculatedChi();
+		uint256 attodaiHrd = balanceOf(tokenHolder);
+		return convertAttodaiHrdToAttodai(attodaiHrd, rontodaiPerPot);
+	}
+
+	function totalSupplyDai() external view returns(uint256 attodai) {
+		uint256 rontodaiPerPot = calculatedChi();
+		attodai = convertAttodaiHrdToAttodai(totalSupply(), rontodaiPerPot);
+		return attodai;
```

Comment:
Style: In the function above you don't inline the `balanceOf` call but you do a direct return, in this one you inline the `totalSupply` call but then do a separate line return.  Is there a reason for the difference in style between these?

My preferenc in this case is probably a mix of the two
```solidity
return convertAttodaiHrdToAttodai(totalSupply(), rontodaiPerPot);
```

Reply:
Trying to lean on maker's style, but i agree. i think the idea is "it's nice to label what you're returning" and then if you don't set it to that variable in the returns() , it feels like you're missing a step?

Reply:
I really like named returns, it makes working with the ABI a lot better.  What I _dislike_ is the fact that you can accidentally not return and thus return an uninitialized variable.  I have lobbied in the past for Solc to have a hard requirement on there being a `return` statement present, even if you provide named returns but so far that hasn't happened.

For this project, I advocate for keeping the named return, but not for assigning to it and leveraging the implicit return value.  I'm fine with what you have in either of these as well though, my preference on the latter is very minor.